### DISCUSSION TITLE
Issue with terragrunt run-all destory

### DISCUSSION THREAD

**Discussion by Aman1994** [gruntwork-io/knowledge-base#531] [Help] [answered]
Hi, We install argocd using oboukili/argocd source and our code has something like 

```terraform
resource "helm_release" "argocd" {
  name             = "argo-cd"
  chart            = "argo-cd"
  repository       = "https://argoproj.github.io/argo-helm"
  create_namespace = true
  namespace        = "argocd"
  version          = "3.29.5"
  values           = [
    file("argocd.yaml")
  ]
  depends_on       = [
    sealedsecret_local.argocd_repos,
  ]
}

resource "kubectl_manifest" "argocd_secret" {
  yaml_body         = sealedsecret_local.argocd_repos.yaml_content
  force_new         = true
  server_side_apply = true
  wait              = true
}

resource "argocd_repository" "argocd_repos" {
  for_each        = var.argocd_repos
  ssh_private_key = file(each.value.ssh_private_key)
  repo            = each.value.url
  depends_on      = [
    helm_release.argocd
  ]
}
```

We deployed the infra using terragrunt and then went fine. Now, when we try to destroy using `terragrunt run-all destroy` we get 

```
│ Error: Failed to init clients
│ 
│   with argocd_repository.argocd_repos["k8id"],
│   on argocd.tf line 49, in resource "argocd_repository" "argocd_repos":
│   49: resource "argocd_repository" "argocd_repos" {
│ 
│ EOF
╵
╷
│ Error: Failed to init clients
│ 
│   with argocd_repository.argocd_repos["k8id-config"],
│   on argocd.tf line 49, in resource "argocd_repository" "argocd_repos":
│   49: resource "argocd_repository" "argocd_repos" {
│ 
│ EOF

```
Any idea what the issue is ?

---

<ins datetime="2022-08-09T07:25:15Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109113">Tracked in ticket #109113</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
It looks like this was resolved in https://github.com/oboukili/terraform-provider-argocd/issues/196, so closing this out.

**Reply by Aman1994** [reply]
Hi @yorinasub17 - actually no - having tried again that was not the solution. So, the issue still persists. Hence i am re-opening the issue

**Reply by yorinasub17** [reply]
In that case, we will need more info to help out here. It's a bit hard to debug this without seeing the folder structure or the Terragrunt config.

My suggestion would be to first be certain that terragrunt is the problem and not terraform/argocd provider. You can do this by running through [the debug flow](https://terragrunt.gruntwork.io/docs/features/debugging/) to cut terragrunt out of the equation, and using `terraform` to manually destroy the resources in the expected order.

At a cursory look, my best guess as to what might be wrong here is that you don't have the dependencies defined such that terragrunt destroys the infra out of order, causing a critical dependency for ArgoCD to authenticate to be removed.
